packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -89,13 +89,21 @@ Future<io.WebSocket> _defaultOpenChannel(String url, {
   io.WebSocket socket;
 
   Future<void> handleError(dynamic e) async {
-    globals.printTrace('Exception attempting to connect to Observatory: $e');
-    globals.printTrace('This was attempt #$attempts. Will retry in $delay.');
-
+    void Function(String) printVisibleTrace = globals.printTrace;
     if (attempts == 10) {
-      globals.printStatus('This is taking longer than expected...');
+      globals.printStatus('Connecting to the Observatory is taking longer than expected...');
+    } else if (attempts == 20) {
+      globals.printStatus('Still attempting to connect to the Observatory...');
+      globals.printStatus(
+        'Consider running with --host-vmservice-port to use a specific port '
```

Comment:
I'd imagine that most users hitting this point are actually dealing with a native crash, so this advice may not be the most useful.

Reply:
consider adding messaging to check native logging (adb, xcode) and ensure the app started correctly on the device

Reply:
I'll change the wording to recommend trying `--host-vmservice-port` if the Flutter application is visible on the device and to check the device logs otherwise.